scripts/globals/survival_guide.lua

```diff
@@ -0,0 +1,200 @@
+require("scripts/globals/settings")
+require("scripts/globals/survival_guide_map")
+
+dsp = dsp or {}
+dsp.survivalGuide = dsp.survivalGuide or {}
+
+-- Determines if the survival guide teleport cost is like if you had a Rhapsody in White key item. Does not affect UI! (Default: 0)
+local SURVIVAL_GUIDE_TELEPORT_COST_GIL = 1000
+local SURVIVAL_GUIDE_TELEPORT_COST_TABS = 50
+
+-- This is used for the NationTeleport save/get
+local travelType = dsp.teleport.type.SURVIVAL
+local cutsceneID = 8500
+
+local optionMap =
+{
+    TELEPORT = 1,
+    UNKNOWN = 2,
+    SET_MENU_LAYOUT = 3,
+    ADD_FAVORITE = 4,
+    REMOVE_FAVORITE = 5,
+    REPLACE_FAVORITE = 6,
+    TELEPORT_MENU = 7
+}
+
+-------------------------------------------------
+-- local functions
+-------------------------------------------------
+
+-- 1 = Original zones
+-- 2 = Zilart Zones
+-- 4 = Chains of Promethia Zones
+-- 8 = Treasures of Aht Urgahn
+-- 16 = Wings of the Goddess
+-- 2048 = Seekers of Auldoin
+local function getEnabledExpansions()
+    -- Original areas and zilart are always available.
+    local returnValue = 3
+
+    if ENABLE_COP == 1 then returnValue = returnValue + 4 end
+
+    if ENABLE_TOAU == 1 then returnValue = returnValue + 8 end
+
+    if ENABLE_WOTG == 1 then returnValue = returnValue + 16 end
+
+    if ENABLE_SOA == 1 then returnValue = returnValue + 2048 end
+
+    return returnValue
```

Comment:
While it's technically possible for these settings to be changed while the server is already booted, I wonder if enabledExpansions should be calculated and defined just once upon the first require of this file (`dsp.survivalGuide.enabledExpansions = getEnabledExpansion()`), and then that value gets passed into the CS each time, instead of the onTrigger having to go through this function every time a player talks to a guide.

Reply:
Fixed